src/CTA.Rules.Actions/ActionHelpers/ServerConfigTemplates.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using CTA.Rules.Models;
+
+namespace CTA.Rules.Actions
+{
+    public static class ServerConfigTemplates
+    {
+        public const string ConfigureServicesMethod = "ConfigureServices";
+        public const string ConfigureMiddlewareMethod = "Configure";
+
+        public static List<string> middlewareConfigMethods = new List<string>()
+        {
+            ConfigureServicesMethod,
+            ConfigureMiddlewareMethod
+        };
+
+        public static Dictionary<string, string> configurationExpressionTemplates = new Dictionary<string, string>()
+        {
+            {WebServerConfigAttributes.Authentication.ToString(), "app.UseAuthentication();"},
+            {WebServerConfigAttributes.Authorization.ToString(), "app.UseAuthorization();" },
+            {WebServerConfigAttributes.Modules.ToString(),"app.UseMiddleware<{0}>();"},
+            {WebServerConfigAttributes.Handlers.ToString(), @"app.MapWhen(context => context.Request.Path.ToString().EndsWith({0}),appBranch => {{ appBranch.UseMiddleware<{1}>();}});" }
+        };
+
+        public static Dictionary<string, string> serviceExpressionTemplates = new Dictionary<string, string>()
+        {
+            {WebServerConfigAttributes.Authorization.ToString(), @"services.AddAuthorization(options =>{options.FallbackPolicy = options.DefaultPolicy;});"},
+            {WebServerConfigAttributes.WindowsAuthentication.ToString(), @"services.AddAuthentication(NegotiateDefaults.AuthenticationScheme).AddNegotiate(options =>{options.Events = new NegotiateEvents(){OnAuthenticationFailed = context => { return Task.CompletedTask}};});" }
```

Comment:
ouch... we can't go in this route, look at this https://docs.microsoft.com/en-us/aspnet/core/security/authentication/windowsauth?view=aspnetcore-5.0&tabs=visual-studio#kerberos-authentication-and-role-based-access-control-rbac ... I would say put something like this 
//Please ensure the scheme is correct, else change your scheme.
services.AddAuthentication(NegotiateDefaults.AuthenticationScheme)
        .AddNegotiate(options =>
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
            {
                options.EnableLdap(settings =>
                {
                    settings.Domain = please enter the domain;
                    settings.MachineAccountName = please enter the machine;
                    settings.MachineAccountPassword = enter the password if authentication is not kerberos, else comment;
                });
            }
        });

Reply:
this way it will generate build error and customer is forced to see and make necessary changes...

Reply:
options should be manually added by user. It will become more tricky if we keep adding more complex expression templates

Reply:
Enable LDAP is compatible in .net 5.0 framework